server/src/main/java/com/epam/aidial/core/server/controller/ResourceController.java

```diff
@@ -164,13 +176,32 @@ private Future<Pair<ResourceItemMetadata, String>> getResourceData(ResourceDescr
         }, false);
     }
 
+    private void validateCustomApplication(Application application) {
```

Comment:
Please move to ApplicationService.

Reply:
can not do this directly due to circular dependencies.
![image](https://github.com/user-attachments/assets/25938159-8a2a-4a03-a4ac-42ec749dc5b2)

Reply:
application service can not be constructed before access service in that scenario

Reply:
But still there is an ugly trick with context -> proxy -> service access

Reply:
moved